sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -257,7 +260,40 @@ internal static bool TryExtractErrorContent(Response response, out ResponseError
                 {
                     return false;
                 }
-                error = System.Text.Json.JsonSerializer.Deserialize<ErrorResponse>(content)?.Error;
+                var reader = new System.Text.Json.Utf8JsonReader(response.Content);
+                int objectDepth = 0;
+                while (reader.Read())
+                {
+                    if (code is not null && message is not null || objectDepth > 2)
+                    {
+                        break;
+                    }
+                    switch (reader.TokenType)
+                    {
+                        case System.Text.Json.JsonTokenType.EndObject:
+                            objectDepth++;
+                            break;
+                        case System.Text.Json.JsonTokenType.PropertyName:
+                            var propName = reader.GetString();
+                            if (!reader.Read())
+                            {
+                                break;
+                            }
+                            if (propName == "message")
+                            {
+                                message = reader.GetString();
+                            }
+                            else if (propName == "code")
```

Comment:
Will this populate InnerError, Target, or Details if they're present?

Reply:
Are you sure?  If you add a test that sets

```
response.ContentStream = GetStream(canSeek, "{ \"error\": { \"code\":\"StatusCode\", \"message\":\"Custom message\", \"target\":\"Target\" }}");
```

I think you can see it set for anything that has access to the internal property.

Reply:
That will definitely add the content to the formatted message, but I don't think it affects the properties of the exception.